infra/presubmit.py

```diff
@@ -0,0 +1,173 @@
+      print('''Error: build.sh contains -lFuzzingEngine on line: {0}.
+Please use $LIB_FUZZING_ENGINE.'''.format(line_num))
+      return False
+  return True
+
+
+class ProjectYamlChecker:
+  def __init__(self, project_yaml_filename):
+    self.project_yaml_filename = project_yaml_filename
+    with open(project_yaml_filename) as file_handle:
+      self.project_yaml = yaml.safe_load(file_handle)
+
+    self.success = True
+
+    self.checks = [self.check_project_yaml_constants,
+                   self.check_required_sections,
+                   self.check_valid_section_names,
+                   self.check_valid_emails]
+
+  def do_checks(self):
+    """Do all project.yaml checks. Return True if they pass."""
+    if self.is_disabled():
+      return True
+    for check_function in self.checks:
+      check_function()
+    return self.success
+
+  def is_disabled(self):
+    """Is this project disabled."""
+    return self.project_yaml.get('disabled', False)
+
+  def print_error_message(self, message, *args):
+    """Print an error message and set self.success to False."""
+    self.success = False
+    message = message % args
+    print('Error in %s: %s' % (self.project_yaml_filename, message))
+
+  def check_project_yaml_constants(self):
+    """Check that certain sections only have certain constant values."""
+    sections_and_constants = {
+        'sanitizers': ['address', 'none', 'memory', 'address'],
+        'architectures': ['i386', 'x86_64'],
+        'engines': ['afl', 'libfuzzer', 'honggfuzz']
+    }
+    success = True
+    for section, constants in sections_and_constants.items():
+      if section not in self.project_yaml:
+        continue
+      section_contents = self.project_yaml[section]
+      for constant in section_contents:
+        if constant not in section_contents:
+          self.print_error_message('%s not one of %s', constant, constants)
+
+  def check_valid_section_names(self):
+    """Check that only valid sections are included."""
+    # Note, this list must be updated when we allow new sections.
+    valid_section_names = [
+        'homepage', 'primary_contact', 'auto_ccs', 'sanitizers',
+        'architectures', 'disabled'
+    ]
+    for name in self.project_yaml:
+      if name not in valid_section_names:
+        self.print_error_message(
+            '%s not a valid section name (%s)', name, valid_section_names)
+
+  def check_required_sections(self):
+    """Check that all required sections are present."""
+    required_sections = ['primary_contact']
+    for section in required_sections:
+      if section not in self.project_yaml:
+        self.print_error_message(
+            'No %s section.', section)
+
+
+  def check_valid_emails(self):
+    """Check that emails are valid looking."""
+    for section in ['auto_ccs', 'primay_contact']:
+      for email_address in self.project_yaml.get(section, []):
+        if not ('@' in email_address and '.' in email_address):
```

Comment:
nit: I think `if '@' not in email_address or '.' not in email_address` is easier to comprehend